### DISCUSSION TITLE
Align HARK Market and dolark terminology?

### DISCUSSION THREAD

**Discussion by sbenthall** [econ-ark/HARK#1672] [General]
The HARK Market class uses the creative terms "reap", "mill", and "sow".

dolark, which is trying to do something similar with its aggregation and equilibrium mechanisms, uses slightly different terminology.
 - it retains the references to exogenous shocks and parameters from dolo
 - it adds references to `aggregate`, `projection`, and `equilibrium`

I wonder if HARK would do well to move in a less metaphorical direction.

One benefit of this is that it helps standardize, conceptually, what the aggregated, "sow" variables are with respect to the agent's problem. dolark (following dolo) gives the agent the market variables via its expanded notion of an exogenous process (which can vary over time; it's not necessarily an IID sample).

From my perspective, this is a question of getting clarity on what's going on:
 - at the mathematical level. What kind of multi-agent process is this, if it is no longer a single-agent MDP? Is there a formal characterization of it with specific terms?
 - at the economic level. What is meant by these functions, in terms of what the models represent?

and letting the answers to those questions drive terminology.

**Comment by llorracc** [comment]
The HARK architecture is drawn from KS.

* Maybe there needs to be a publication about the general method. (besides the original HARK paper)
* Form a standards committee to settle on this sort of thing.
